js/client-battle.js

```diff
@@ -1002,18 +997,23 @@
 				text += '</div></div>';
 				break;
 
-			case 'pokemon':
-				var pokemon = this.battle.getPokemon(thing);
+			case 'pokemon': // Tooltip for active pokemon. Initially we select the spectator/opponent sees view.
+				var side = this.battle[thing.slice(0, -1) + "Side"];
+				var pokemon = side.active[thing.slice(-1)];
 				if (!pokemon) return;
 				/* falls through */
-			case 'sidepokemon':
+			case 'sidepokemon': // Tooltip for switching pokemon; myPokemon contains the latest server state.
 				var battlePokemon;
 				var myPokemon;
 				if (this.myPokemon) {
 					if (!pokemon) {
+						// We didn't fall through, so we're switching. Just show the server state.
 						pokemon = this.myPokemon[parseInt(thing)];
 						battlePokemon = this.battle.getPokemon('other: old: ' + pokemon.ident, pokemon.details);
 					} else if (pokemon.side === this.battle.mySide) {
```

Comment:
I guess your following code is carefully designed to prevent this situation.

Still, though, I don't like the complication that results from mixing information sources like this, I prefer for information sources to be clearer.